### ISSUE TITLE
DPI: porting to kernel 3.10.0-514.el7.x86_64

### ISSUE THREAD

**Issue by gsanchietti** [NethServer/dev#5170]
CentOS 7.3 comes with the new kernel 3.10.0-514.
This release can be used to build the xt_ndpi module which actually works only on kernel-lt.

Refactor the current implementation to work with CentOS 7.3 standard kernel:
- [x] rebuild kmod-xt_ndpi
- [x] remove kernel-lt dependency
- [x] remove `/etc/sysconfig/kernel` template

**Comment by gsanchietti** [comment]
Before upgrading current installations where nethserver-ndpi is already installed, execute the following commands to allow the configuration of new kernels:
```
rm -f /etc/sysconfig/kernel 
yum reinstall grubby
```

**Comment by gsanchietti** [comment]
In nethserver-testing:
- nethserver-ndpi-1.0.0-1.2.gb20aa4e.ns7.noarch.rpm
- kmod-xt_ndpi-2.0.0-1.ns7.x86_64.rpm

Before testing, make sure *cr* repository is enabled:
```
yum --enablerepo=cr,nethserver-testing install nethserver-ndpi kmod-xt_ndpi
```
*Test case 1: clean install*
- Install the new packages on a clean RC2 machine with at least one red interface
- Reboot the machine to activate the kernel from cr repo (this step will not be required when installing on a RC 3)
- Create a block rule using a DPI protocol from the "Firewall rules" page
- Check the firewall blocks the selected network traffic

*Test case 2: update*
- Prepare the machine  where nethserver-ndpi is already installed:
```
rm -f /etc/sysconfig/kernel 
yum reinstall grubby
```
- Install the updated packages
- Reboot
- Check the new kernel is `3.10.0-514.el7.x86_64`: `uname -r`
- Check ndpi module is loaded: `lsmod | grep xt_ndpi`
- Create a block rule using a DPI protocol from the "Firewall rules" page
- Check the firewall blocks the selected network traffic

You can free space in updated machines by removing the kernel-lt package:
```
yum remove kernel-lt
```

**Comment by filippocarletti** [comment]
Test case 2: OK

**Comment by filippocarletti** [comment]
Test case 1: FAILED

History:
Followed instructions to install packages: no errors. The dashboard asked to reboot to activate nDPI, reboot and the yellow todo disappeared.
I can't create rules using nDPI protocols.
The DPI status page is empty.
xt_ndpi kmod is not loaded.
I loaded it with modprobe xt_ndpi and everything works as expected.

**Comment by gsanchietti** [comment]
New RPM with weak-modules support:
nethserver-ndpi-1.0.0-1.3.gbcd00f5.ns7.noarch.rpm

**Comment by filippocarletti** [comment]
nethserver-ndpi-1.0.0-1.3.gbcd00f5 has fixed the problem.

**Comment by gsanchietti** [comment]
Released in 7.3.1611/base:
- nethserver-ndpi-1.1.0-1.ns7.noarch.rpm
- kmod-xt_ndpi-2.0.0-1.ns7.x86_64.rpm
